File: DevExpress-Demo/VB/RibbonDemo.Wpf/Modules/RibbonSimplePad/TemplateSelectors/InlineImageContentTemplateSelector.vb
Imports System.Windows
Imports System.Windows.Controls

Namespace RibbonDemo

    Public Class InlineImageContentTemplateSelector
        Inherits DataTemplateSelector

        Public Property Dictionary As TemplateSelectorDictionary

        Public Overrides Function SelectTemplate(ByVal item As Object, ByVal container As DependencyObject) As DataTemplate
            Return Dictionary(CType(item, InlineImageBorderType).ToString())
        End Function
    End Class
End Namespace

File: DevExpress-Demo/CS/RibbonDemo.Wpf/Modules/RibbonSimplePad/TemplateSelectors/InlineImageContentTemplateSelector.cs
using System.Windows;
using System.Windows.Controls;
namespace RibbonDemo {
    public class InlineImageContentTemplateSelector : DataTemplateSelector {
        public TemplateSelectorDictionary Dictionary { get; set; }
        public override DataTemplate SelectTemplate(object item, DependencyObject container) {
            return Dictionary[((InlineImageBorderType)item).ToString()];
        }
    }
}
